client.go

```diff
@@ -85,13 +85,23 @@ func (c *Client) sendRequest(req *http.Request, v interface{}) error {
 		return fmt.Errorf("error, status code: %d, message: %w", res.StatusCode, errRes.Error)
 	}
 
-	if v != nil {
-		if err = json.NewDecoder(res.Body).Decode(v); err != nil {
+	return decodeResponse(v, *res)
+}
+
+func decodeResponse(v interface{}, res http.Response) error {
```

Comment:
```suggestion
func decodeResponse(v any, res http.Response) error {
```

Reply:
Could we also rename `v` to something sensible here, maybe `response`?

Reply:
I have:
* Used `any`
* Updated `decodeResponse` arguments to `body io.Reader, v any`
  * This ordering is a little nicer (input/output)
  * I struggled to come up with a good name for `v` - this is what is used in the `json` library to mean any value that we will decode into... so I feel like it is a better name than response which is used by `http` (at least in this context)